src/parachain/issue.ts

```diff
@@ -175,25 +136,20 @@ export class DefaultIssueAPI implements IssueAPI {
         if (!vaultId) {
             vaultId = await this.vaultsAPI.selectRandomVaultIssue(amountSat);
         }
-        const griefingCollateralPlanck = await this.getGriefingCollateralInPlanck(amountSat.toString());
-        const requestIssueTx = this.api.tx.issue.requestIssue(amountSat, vaultId, griefingCollateralPlanck);
-        const result = await this.transaction.sendLogged(requestIssueTx, this.account);
-        if (!this.isRequestSuccessful(result.events)) {
-            Promise.reject("Request failed");
-        }
-
+        const griefingCollateralPlanck = await this.getGriefingCollateralInPlanck(amountSat);
+        const requestIssueTx = this.api.tx.issue.requestIssue(amountSat, vaultId, griefingCollateralPlanck.toString());
+        const result = await this.transaction.sendLogged(requestIssueTx, this.account, this.api.events.issue.RequestIssue);
         const id = this.getIssueIdFromEvents(result.events);
         const issueRequest = await this.getRequestById(id);
         return { id, issueRequest };
     }
 
-    async execute(issueId: H256, txId: H256Le, merkleProof: Bytes, rawTx: Bytes): Promise<boolean> {
+    async execute(issueId: H256, txId: H256Le, merkleProof: Bytes, rawTx: Bytes): Promise<void> {
         if (!this.account) {
             throw new Error("cannot request without setting account");
         }
         const executeIssueTx = this.api.tx.issue.executeIssue(issueId, txId, merkleProof, rawTx);
-        const result = await this.transaction.sendLogged(executeIssueTx, this.account);
-        return this.isExecutionSuccessful(result.events);
+        await this.transaction.sendLogged(executeIssueTx, this.account, this.api.events.issue.ExecuteIssue);
```

Comment:
Is the Promise rejected if unsuccessful?

Reply:
It is rejected here to avoid duplication: https://github.com/interlay/polkabtc-js/blob/dan/client-transition-to-api/src/utils/transaction.ts#L63